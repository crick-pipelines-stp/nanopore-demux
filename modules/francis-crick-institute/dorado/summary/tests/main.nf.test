nextflow_process {

    name "Test Module DORADO_SUMMARY"
    script "../main.nf"
    process "DORADO_SUMMARY"

    tag "modules"
    tag "ont"
    tag "ont_dorado"
    tag "ont_dorado/summary"

    test("summary") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test' ],
                    file('$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed/1000_reads_6_samples.bam', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match()}
            )
        }
    }
}
