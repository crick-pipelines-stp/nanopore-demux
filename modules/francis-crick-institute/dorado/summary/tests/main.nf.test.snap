{
    "summary": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tsv:md5,462aa9608de3cbadc17a7e9141db12c3"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b6e8fc68eadc183c1f8dbec9e48d79b5"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tsv:md5,462aa9608de3cbadc17a7e9141db12c3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b6e8fc68eadc183c1f8dbec9e48d79b5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:31:16.068475"
    }
}